### PR TITLE
Hide unneccesary authenticate popup block

### LINKED ISSUE #29208: [Issue] Authentication popup should be removed from Checkout page.
This issue is automatically created based on existing pull request: magento/magento2#29183: Authentication popup should be removed from Checkout page.

---------

### Preconditions (*)

Magento is rendering \Magento\Customer\Block\Account\AuthenticationPopup block on every frontend page, while this block is used only for authorizing guest users before accessing Checkout. So it's totally useless on the Checkout page and may cause some problems for developers.

### Related Pull Requests

None

### Fixed Issues (if relevant)

1. None

### Steps to reproduce (*)

1. Disable Guest Checkout ( Go to **Admin -> Stores -> Configuration -> Sales -> Checkout** and set **Allow Guest Checkout** to **No**);
2. Go to Storefront;
3. Make sure code `<div id="authenticationPopup"...` is present in any storefront page source code;

**Actual Result:** :heavy_check_mark: `authenticationPopup` is present in any storefront page

4. Login As Customer, proceed to Checkout;
5. Make sure code `<div id="authenticationPopup"...` is absent in Checkout page source code.

**Actual Result:** :heavy_multiplication_x: Code `<div id="authenticationPopup"...`block  is present in Checkout page source code.

**Expected Result:** :heavy_check_mark: Code `<div id="authenticationPopup"...` should be absent in Checkout page source code.

Screenshot from 2020-07-20 12-06-42

Checkout functionality still works as previously (expected).

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Default layout magento have render authenticate-popup block in all pages. This can impact performance a little bit. For now it's not affect much to store. In near future we probably need refactor it
Here is the modal look like when owner disabled guest checkout and from storefront user click checkout button in minicart
Screenshot from 2020-10-25 17-54-42

When this modal active it will show for user can login/create account
In checkout we don't need this popup because we have "Sign in" link in top right of page, so we can safe hide it or even remove it. For backward compatible (previous magento version and 3rd-party modules) we should hide it instead completely remove it

Screenshot from 2020-10-25 15-36-02

### Fixed Issues (if relevant)

1. Fixes magento/magento2#29208

### Manual testing scenarios (*)

1. ...
2. ...

Before in all pages will have authentication popup
Screenshot from 2020-10-25 15-29-12

After change html render in checkout
Screenshot from 2020-10-25 15-39-13

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by rogyar** [comment]
Hi @mrtuvn. Thanks for your fix. I thought it has been removed a long time ago :)  
Could you cover this change with an MFTF test, please? The test will add a product to the shopping cart, go to the checkout and make sure that there's no authentication popup selector present on the page. 

Thank you!

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@rogyar can you give me a review

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run all tests

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontCheckoutVerifyAuthenticatePopupNotVisible.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            </createData>
+        </before>
+        <after>
+            <!--Delete created data-->
+            <deleteData createDataKey="createSimpleProduct" stepKey="deleteSimpleProduct"/>
+            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
+        </after>
+
+        <!--Go to product page-->
+        <amOnPage url="$$createSimpleProduct.custom_attributes[url_key]$$.html" stepKey="navigateToSimpleProductPage"/>
```
Comment (rogyar) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontCheckoutVerifyAuthenticatePopupNotVisible.xml]: Great job. The last thing I would ask you to do is to utilize the existing action groups for such actions. 

https://github.com/magento/magento2/blob/0262f1428561f5c329a50bfbd62714fa58e175be/app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontOpenProductPageActionGroup.xml#L10

Thank you!

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by mrtuvn** [comment]
@rogyar Updated PR as your suggestions

**Comment by rogyar** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests CE, Functional Tests B2B, Functional Tests EE

**Comment by rogyar** [comment]
Hi @mrtuvn there's something wrong with removing entities after the test execution. Could you try to swap removing product and category in the `after` section, please?

**Comment by mrtuvn** [comment]
@rogyar updated my PR for mftf test

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Got broken issue with mftf "Entity with a CreateDataKey of createProduct could not be found"
Any reason why this happen @rogyar @engcom-Charlie @engcom-Alfa

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontCheckoutVerifyAuthenticatePopupNotVisibleTest.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        </before>
+        <after>
+            <!--Delete created data-->
+            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
+            <deleteData createDataKey="createSimpleProduct" stepKey="deleteSimpleProduct"/>
+        </after>
+
+        <!--Go to product page-->
+        <actionGroup ref="StorefrontOpenProductPageActionGroup" stepKey="openStorefrontProductPage">
+            <argument name="productUrl" value="$$createProduct.custom_attributes[url_key]$$"/>
```
Comment (rogyar) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontCheckoutVerifyAuthenticatePopupNotVisibleTest.xml]: I believe the possible reason for that -  we use `$$createProduct` but in the "before" section we use `createSimpleProduct` as a key.

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
merged 2.4-develop branch + rebased

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8513](https://jira.corp.magento.com/browse/ENGCOM-8513) has been created to process this Pull Request

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
i think this one may resolve impact with performance for checkout page. Also improve customer checkout experience. Can we consider increase to P3 priority ?
@gabrieldagama @rogyar

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Pull request closed without merging**